src/duckling/project/map-dimension.service.ts

```diff
@@ -0,0 +1,23 @@
+import {Injectable} from '@angular/core';
+
+import {Vector} from '../math/vector';
+
+/**
+ * Service that holds the dimension and grid information for the active map
+ */
+@Injectable()
+export class MapDimensionService {
+    /**
+     * Width and height of active map
+     */
+    dimension : Vector;
```

Comment:
This should probably be in the store. You could replace Project.currentMap with an object that has the map's key, version, dimension, and version. Alternatively, you could remove currentMap from Project and create a new MapService that has grid size information.